src/entry-client.js

```diff
@@ -42,16 +47,18 @@ export default function initializeClient(createApp, clientOpts) {
         // Allow a function to be passed that can generate a route-aware
         // module name
         const getModuleName = (c, route) => (
-            typeof c.vuex.moduleName === 'function' ?
+            isFunction(c.vuex.moduleName) ?
                 c.vuex.moduleName({ $route: route }) :
                 c.vuex.moduleName
         );
 
         // Before routing, register any dynamic Vuex modules for new components
         router.beforeResolve((to, from, next) => {
             try {
+                const fetchDataArgs = { app, route: to, router, store, from };
                 router.getMatchedComponents(to)
                     .filter(c => 'vuex' in c)
+                    .filter(c => !shouldIgnoreRouteUpdate(c, fetchDataArgs))
```

Comment:
We should never need to register a component's vuex module on ignored route updates since it will always register on the first update which should never be ignored